topology/src/main/java/io/camunda/zeebe/topology/api/TopologyRequestServer.java

```diff
@@ -104,4 +117,26 @@ private void registerGetTopologyQueryHandler() {
         request -> topologyManagementApi.getTopology().toCompletableFuture(),
         serializer::encode);
   }
+
+  private CompletableFuture<Either<ErrorResponse, TopologyChangeResponse>> mapResponse(
+      final ActorFuture<TopologyChangeResponse> topologyManagementApi) {
+    return topologyManagementApi
+        .toCompletableFuture()
+        .thenApply(Either::<ErrorResponse, TopologyChangeResponse>right)
+        .exceptionally(this::mapError);
```

Comment:
:x: I think this does not work correctly but I'm not sure why. If I try this version and an exception is thrown in `OperationApplier#init`, the client gets "500 Internal Server Error". If I change this code to use `handle` (instead of `thenApply` followed with `exceptionally`), I get the expected "400 Bad Request"/

Reply:
Ah, this seems to be because `exceptionally` receives the exception wrapped in a `CompletionException` :roll_eyes:

Reply:
Nice catch.